tests/all_tests/produce_fetch.rs

```diff
@@ -196,9 +196,11 @@ fn fetch_records(
     );
 
     let mut fetched_records = partition_response.records.clone().unwrap();
```

Comment:
Would be nice to have a test covering the new wrapped version but not a big deal.